Src/Newtonsoft.Json/Linq/JObject.cs

```diff
@@ -828,7 +841,15 @@ EventDescriptorCollection ICustomTypeDescriptor.GetEvents()
         /// </returns>
         protected override DynamicMetaObject GetMetaObject(Expression parameter)
         {
+#if HAVE_APPCONTEXT
+            if (!MiscellaneousUtils.DynamicIsSupported)
```

Comment:
Should this (and similar) be:

```suggestion
#if HAVE_DYNAMIC
            if (!MiscellaneousUtils.DynamicIsSupported)
```

?